### PR TITLE
Add a sample app for using OpenTelemetry Java Extension

### PR DESCRIPTION
**Pull request by heyams**

### CODE REVIEW

File: opentelemetry-api/java-agent/extensions/CustomizeLogRecordExporter/src/main/java/com/example/DemoAutoConfigurationCustomizerProvider.java [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+ * This is one of the main entry points for Instrumentation Agent's customizations. It allows
+ * configuring the {@link AutoConfigurationCustomizer}. See the {@link
+ * #customize(AutoConfigurationCustomizer)} method below.
+ *
+ * <p>Also see https://github.com/open-telemetry/opentelemetry-java/issues/2022
+ *
+ * @see AutoConfigurationCustomizerProvider
+ */
+@AutoService(AutoConfigurationCustomizerProvider.class)
+public class DemoAutoConfigurationCustomizerProvider
```
Comment (heyams) [inline opentelemetry-api/java-agent/extensions/CustomizeLogRecordExporter/src/main/java/com/example/DemoAutoConfigurationCustomizerProvider.java:25]: @trask do we need to use DemoAutoConfigurationCustomizerProvider in order to apply the new exporter? 
This setup didn't invoke the new DemoLogRecordExporter somehow.

File: opentelemetry-api/java-agent/extensions/CustomizeLogRecordExporter/src/main/java/com/example/DemoLogRecordExporter.java [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+    public DemoLogRecordExporter(LogRecordExporter delegate) {
+        this.delegate = delegate;
+    }
+    @Override
+    public CompletableResultCode export(Collection<LogRecordData> collection) {
+        List<LogRecordData> copiedList = new ArrayList<>();
+        for(LogRecordData data : collection) {
+            String pidValue = data.getAttributes().get(AttributeKey.stringKey(TARGET_ATTRIBUTE_KEY));
+            logger.debug("###### pidValue: " + pidValue);
```
Comment (heyams) [inline opentelemetry-api/java-agent/extensions/CustomizeLogRecordExporter/src/main/java/com/example/DemoLogRecordExporter.java:29]: debug logs are to be removed later.

File: opentelemetry-api/java-agent/extensions/CustomizeLogRecordExporter/src/main/java/com/example/DemoAutoConfigurationCustomizerProvider.java [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+    implements AutoConfigurationCustomizerProvider {
+
+    @Override
+    public void customize(AutoConfigurationCustomizer autoConfiguration) {
+
```
Comment (trask) [inline opentelemetry-api/java-agent/extensions/CustomizeLogRecordExporter/src/main/java/com/example/DemoAutoConfigurationCustomizerProvider.java:30]: if you add a `System.out.println` here, does it get called?

File: opentelemetry-api/java-agent/extensions/CustomizeLogRecordExporter/src/main/java/com/example/DemoAutoConfigurationCustomizerProvider.java [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    }
+
+    private Map<String, String> getDefaultProperties() {
+        Map<String, String> properties = new HashMap<>();
+        properties.put("otel.exporter.otlp.endpoint", "http://backend:8080");
+        properties.put("otel.exporter.otlp.insecure", "true");
+        properties.put("otel.config.max.attrs", "16");
+        properties.put("otel.traces.sampler", "demo");
+        return properties;
+    }
```
Comment (trask) [inline opentelemetry-api/java-agent/extensions/CustomizeLogRecordExporter/src/main/java/com/example/DemoAutoConfigurationCustomizerProvider.java:43]:
```suggestion
```

File: opentelemetry-api/java-agent/extensions/CustomizeLogRecordExporter/src/main/java/com/example/DemoAutoConfigurationCustomizerProvider.java (hunk 3 above)
Comment (heyams) [reply opentelemetry-api/java-agent/extensions/CustomizeLogRecordExporter/src/main/java/com/example/DemoAutoConfigurationCustomizerProvider.java:30]: no

**Pull request closed without merging**